src/MapkitProvider.tsx

```diff
@@ -0,0 +1,56 @@
+/* global mapkit */
+
+import React from 'react'
+import load from 'little-loader'
+
+type MapkitContextType = {
+  isInProvider: boolean
+  mapkit: typeof mapkit | undefined
+}
+
+export const MapkitContext = React.createContext<MapkitContextType>({
+  isInProvider: false,
+  mapkit: undefined,
+})
+
+type ProviderProps = {
+  // ⚠️ Pick between callbackUrl or token.
+  // https://developer.apple.com/documentation/mapkitjs/mapkit/2974045-init
+  tokenOrCallback: string
+}
+
+export const MapkitProvider: React.FC<ProviderProps> = ({
+  tokenOrCallback,
+  children,
+}) => {
+  const existingContext = React.useContext(MapkitContext)
+
+  const [context, setContext] = React.useState<MapkitContextType>({
+    mapkit: existingContext.mapkit,
+    isInProvider: true,
+  })
+
+  React.useEffect(() => {
+    if (!existingContext.isInProvider) {
```

Comment:
Why do you need a flag here for `isInProvider` to do these things?

Reply:
I only want to setup context if there isn’t already one above, this should handle that check